### PR TITLE
Rename S32 to NXP S32

### PR DESCRIPTION
**Pull request by manuargue**
Following updates previously done for other drivers and as requested by reviewers (e.g. https://github.com/zephyrproject-rtos/zephyr/pull/51451#issuecomment-1294384579), rename all occurrences of S32 to NXP S32 to avoid ambiguity.

### CODE REVIEW

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
LGTM once the HAL change is merged.

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by manuargue** [comment]
@dleach02 @danieldegrasse could you pls have a look at this patch? thanks

**Review by dleach02 — APPROVED** [review]
@manuargue, we need to coordinate this merge as there are a number of S32 changes. 

ping me so we can do this.

**Pull request merged**
